stonesoup/feeder/filter.py

```diff
@@ -154,17 +170,17 @@ def __init__(self, *args, **kwargs):
             self.mapping = tuple(range(len(self.limits)))
 
     @BufferedGenerator.generator_method
-    def detections_gen(self):
+    def data_gen(self):
         num_dims = len(self.limits)
-        for time, detections in self.detector:
-            outlier_detections = set()
-            for detection in detections:
-                state_vector = detection.state_vector
+        for time, states in self.reader:
+            outlier_data = set()
+            for state in states:
+                state_vector = state.state_vector
```

Comment:
Not too sure if this is going to work with ``GroundTruthReader``s. These yield a list ``GroundTruthPath`` objects, where each path is an iterable over multiple states. Therefore in this case ``state`` would actually be of type ``GroundTruthPath``, where the actual ``State`` object should be queried as ``state.state`` if I am not mistaken.

Reply:
Calling path.state_vector returns the state vector of the most recent state in the path, so this shouldn't be an issue in the bounding box reducer.

Reply:
Ahh, that's interesting, I wasn't aware of this. Very well, just disregard this comment then 👍